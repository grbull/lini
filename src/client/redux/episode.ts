/* eslint-disable no-param-reassign */
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

import { EpisodeDto } from '../../server/episode/episode.dto';
import { api } from '../utils/api';

type EpisodeState = {
  isLoading: boolean;
  error?: string;
  data?: EpisodeDto;
};

const initialState: EpisodeState = {
  isLoading: false,
};

export const get = createAsyncThunk<EpisodeDto, number>('episode/get', (id) =>
  api.episode.get(id)
);

export const episode = createSlice({
  name: 'episode',
  initialState,
  reducers: {
    clear: (state) => {
      state.data = undefined;
    },
  },
  extraReducers: (builder) => {
    builder.addCase(get.pending, (state) => {
      state.isLoading = true;
    });
    builder.addCase(get.fulfilled, (state, { payload }) => {
      state.isLoading = false;
      state.data = payload;
    });
    builder.addCase(get.rejected, (state, { error }) => {
      state.isLoading = false;
      state.error = error.message;
    });
  },
});

export const episodeActions = { ...episode.actions, get };
