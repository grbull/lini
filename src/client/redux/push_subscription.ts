/* eslint-disable no-param-reassign */
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

import { PushSubscriptionCreateDto } from '../../server/push_subscription/push_subscription.dto';
import { api } from '../utils/api';

type PushSubscriptionState = {
  status: 'loading' | 'idle' | 'error';
  error?: string;
};

const initialState: PushSubscriptionState = {
  status: 'idle',
};

export const create = createAsyncThunk<void, PushSubscriptionCreateDto>(
  'push_subscription/get',
  (createDto) => api.push.create(createDto)
);

export const pushSubscription = createSlice({
  name: 'push_subscription',
  initialState,
  reducers: {},
  extraReducers: (builder) => {
    builder.addCase(create.pending, (state) => {
      state.status = 'loading';
      state.error = undefined;
    });
    builder.addCase(create.fulfilled, (state) => {
      state.status = 'idle';
    });
    builder.addCase(create.rejected, (state, { error }) => {
      state.status = 'error';
      state.error = error.message;
    });
  },
});

export const pushSubscriptionActions = {
  ...pushSubscription.actions,
  create,
};
