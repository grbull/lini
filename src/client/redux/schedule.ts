/* eslint-disable no-param-reassign */
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

import {
  EpisodeDto,
  EpisodeScheduleDto,
} from '../../server/episode/episode.dto';
import { api } from '../utils/api';

type ScheduleState = {
  status: 'init' | 'loading' | 'idle' | 'error';
  data: {
    future: EpisodeDto[];
    past: EpisodeDto[];
  };
  error?: string;
};

const initialState: ScheduleState = {
  status: 'init',
  data: {
    future: [],
    past: [],
  },
};

export const get = createAsyncThunk<EpisodeScheduleDto>('schedule/get', () =>
  api.episode.getSchedule()
);

export const schedule = createSlice({
  name: 'schedule',
  initialState,
  reducers: {},
  extraReducers: (builder) => {
    builder.addCase(get.pending, (state) => {
      state.status = 'loading';
      state.error = undefined;
    });
    builder.addCase(get.fulfilled, (state, { payload }) => {
      state.status = 'idle';
      state.data = payload;
    });
    builder.addCase(get.rejected, (state, { error }) => {
      state.status = 'error';
      state.error = error.message;
    });
  },
});

export const scheduleActions = {
  ...schedule.actions,
  get,
};
