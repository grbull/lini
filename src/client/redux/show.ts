/* eslint-disable no-param-reassign */
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

import { ShowWithEpisodesDto } from '../../server/show/show.dto';
import { api } from '../utils/api';

type ShowState = {
  status: 'init' | 'loading' | 'idle' | 'error';
  data?: ShowWithEpisodesDto;
  error?: string;
};

const initialState: ShowState = {
  status: 'init',
};

export const get = createAsyncThunk<ShowWithEpisodesDto, number>(
  'show/get',
  (id) => api.show.get(id)
);

export const show = createSlice({
  name: 'show',
  initialState,
  reducers: {
    reset: () => {
      return initialState;
    },
  },
  extraReducers: (builder) => {
    builder.addCase(get.pending, (state) => {
      state.status = 'loading';
      state.data = undefined;
      state.error = undefined;
    });
    builder.addCase(get.fulfilled, (state, { payload }) => {
      state.status = 'idle';
      state.data = payload;
    });
    builder.addCase(get.rejected, (state, { error }) => {
      state.status = 'error';
      state.error = error.message;
    });
  },
});

export const showActions = {
  ...show.actions,
  get,
};
