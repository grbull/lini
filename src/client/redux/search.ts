/* eslint-disable no-param-reassign */
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

import { ShowDto } from '../../server/show/show.dto';
import { api } from '../utils/api';

type SearchState = {
  isLoading: boolean;
  error?: string;
  data?: ShowDto[];
};

const initialState: SearchState = {
  isLoading: false,
};

export const query = createAsyncThunk<ShowDto[], string>(
  'search/query',
  (query) => api.show.search(query)
);

export const search = createSlice({
  name: 'search',
  initialState,
  reducers: {
    clear: (state) => {
      state.data = undefined;
    },
  },
  extraReducers: (builder) => {
    builder.addCase(query.pending, (state) => {
      state.isLoading = true;
    });
    builder.addCase(query.fulfilled, (state, { payload }) => {
      state.isLoading = false;
      state.data = payload;
    });
    builder.addCase(query.rejected, (state, { error }) => {
      state.isLoading = false;
      state.error = error.message;
    });
  },
});

export const searchActions = { ...search.actions, query };
