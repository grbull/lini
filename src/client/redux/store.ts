import {
  combineReducers,
  configureStore,
  getDefaultMiddleware,
} from '@reduxjs/toolkit';
import {
  FLUSH,
  PAUSE,
  PERSIST,
  persistReducer,
  persistStore,
  PURGE,
  REGISTER,
  REHYDRATE,
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';

import { episode } from './episode';
import { pushSubscription } from './push_subscription';
import { schedule } from './schedule';
import { search } from './search';
import { show } from './show';
import { subscription } from './subscription';
import { user } from './user';

export const reducers = combineReducers({
  user: user.reducer,
  search: search.reducer,
  subscription: subscription.reducer,
  show: show.reducer,
  episode: episode.reducer,
  schedule: schedule.reducer,
  pushSubscription: pushSubscription.reducer,
});

const persistConfig = {
  key: 'root',
  storage,
  version: 2,
};

const persistedReducer = persistReducer(persistConfig, reducers);

export const store = configureStore({
  reducer: persistedReducer,
  // https://github.com/rt2zz/redux-persist/issues/988#issuecomment-552242978
  middleware: getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }),
  devTools: process.env.NODE_ENV === 'development',
});

export const persistor = persistStore(store);

export type RootState = ReturnType<typeof store.getState>;
