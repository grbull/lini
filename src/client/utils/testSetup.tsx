import { configureStore } from '@reduxjs/toolkit';
import { render, RenderResult } from '@testing-library/react';
import { createMemoryHistory, MemoryHistory } from 'history';
import React, { ReactNode } from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';

import { reducers, RootState } from '../redux/store';

export interface TestSetupProps {
  history?: MemoryHistory;
  state?: Partial<RootState>;
}

export interface TestSetupResult extends RenderResult {
  dispatch: typeof jest.fn;
  history: MemoryHistory<unknown>;
  scrollTo: typeof jest.fn;
}

export function testSetup(
  children: ReactNode,
  props?: TestSetupProps
): TestSetupResult {
  const history = props?.history || createMemoryHistory();
  const store = configureStore({
    reducer: reducers,
    preloadedState: props?.state || undefined,
  });

  // const reduxHistory = [];
  // const dispatch = (action: any): any => reduxHistory.push(action);
  const dispatch = jest.fn();
  store.dispatch = dispatch;

  const scrollTo = jest.fn();
  window.scrollTo = scrollTo;

  const utils = render(
    <Router history={history}>
      <Provider store={store}>{children}</Provider>
    </Router>
  );

  return { dispatch, history, scrollTo, ...utils };
}
