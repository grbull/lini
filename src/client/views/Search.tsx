import cn from 'classnames';
import React, { ReactElement } from 'react';
import { useSelector } from 'react-redux';

import { ErrorMessage } from '../components/ErrorMessage';
import { PageHeader } from '../components/PageHeader';
import { SearchInput } from '../components/SearchInput';
import { ShowGrid } from '../components/ShowGrid';
import { ShowGridSkeleton } from '../components/ShowGridSkeleton';
import { RootState } from '../redux/store';

export function Search(): ReactElement {
  const search = useSelector((state: RootState) => state.search);

  if (search.status === 'error') {
    return (
      <>
        <PageHeader title="Error" />
        <ErrorMessage>{search.error}</ErrorMessage>
      </>
    );
  }

  if (search.status === 'loading') {
    return (
      <>
        <PageHeader title="Library" />
        <SearchInput />
        <ShowGridSkeleton />
      </>
    );
  }

  if (!search.data) {
    return (
      <>
        <PageHeader title="Library" />
        <SearchInput />
        <p className={cn('px-2.5')}>No results to display.</p>
      </>
    );
  }

  return (
    <>
      <PageHeader title="Search" />
      <SearchInput />
      <ShowGrid shows={search.data} />
    </>
  );
}
