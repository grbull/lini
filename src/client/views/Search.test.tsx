/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom/extend-expect';

import React from 'react';

import { ShowDto } from '../../server/show/show.dto';
import { RootState } from '../redux/store';
import { testSetup } from '../utils/testSetup';
import { Search } from './Search';

const showDto: ShowDto = {
  id: 216,
  name: 'Rick and Morty',
  type: 'Animation',
  language: 'English',
  genres: ['Comedy', 'Adventure', 'Science-Fiction'],
  status: 'Running',
  runtime: 30,
  runtimeAverage: 30,
  datePremiered: '2013-12-02',
  officialSite: 'http://www.adultswim.com/videos/rick-and-morty',
  scheduleTime: '23:00:00',
  scheduleDays: ['Sunday'],
  rating: 9,
  imageMedium: '/uploads/images/medium_portrait/1/3603.jpg',
  imageOriginal: '/uploads/images/original_untouched/1/3603.jpg',
  summary:
    'Rick is a mentally gifted, but sociopathic and alcoholic scientist and a grandfather to Morty; an awkward, impressionable, and somewhat spineless teenage boy. Rick moves into the family home of Morty, where he immediately becomes a bad influence.',
  network: {
    id: 10,
    name: 'Adult Swim',
    country: {
      name: 'United States',
      code: 'US',
      timezone: 'America/New_York',
    },
  },
  webChannel: null,
};

describe('Search View', () => {
  it('matches the snapshot', () => {
    const initialState: Partial<RootState> = {
      search: {
        status: 'init',
      },
    };

    const { asFragment } = testSetup(<Search />, { state: initialState });

    expect(asFragment()).toMatchSnapshot();
  });

  it('matches the snapshot when loading', () => {
    const initialState: Partial<RootState> = {
      search: {
        status: 'loading',
      },
    };

    const { asFragment } = testSetup(<Search />, { state: initialState });

    expect(asFragment()).toMatchSnapshot();
  });

  it('matches the snapshot when error', () => {
    const initialState: Partial<RootState> = {
      search: {
        status: 'error',
        error: 'Generic error',
      },
    };

    const { asFragment } = testSetup(<Search />, { state: initialState });

    expect(asFragment()).toMatchSnapshot();
  });

  it('matches the snapshot with no data', () => {
    const initialState: Partial<RootState> = {
      search: {
        status: 'idle',
        data: [],
      },
    };

    const { asFragment } = testSetup(<Search />, { state: initialState });

    expect(asFragment()).toMatchSnapshot();
  });

  it('matches the snapshot with data', () => {
    const initialState: Partial<RootState> = {
      search: {
        status: 'idle',
        data: [showDto],
      },
    };

    const { asFragment } = testSetup(<Search />, { state: initialState });

    expect(asFragment()).toMatchSnapshot();
  });
});
