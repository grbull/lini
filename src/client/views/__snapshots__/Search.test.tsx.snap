// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search View matches the snapshot 1`] = `
<DocumentFragment>
  <header
    class="w-full h-12 flex items-center p-2 md:hidden justify-between"
  >
    <button
      class="w-11 float-left"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-arrow-left fa-w-14 m-1 text-lg"
        data-icon="arrow-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
          fill="currentColor"
        />
      </svg>
    </button>
    <h1
      class="mx-2 font-medium text-lg overflow-x-auto whitespace-nowrap"
    >
      Search
    </h1>
    <div
      class="w-11"
    />
  </header>
  <div
    class="px-2"
  >
    <input
      aria-label="search-input"
      class="w-full mb-4 p-1 text-xl text-center bg-gray-200 dark:bg-gray-600 border border-solid border-gray-800 focus:border-gray-400"
      placeholder="Search..."
      type="search"
      value=""
    />
  </div>
  <div
    class="absolute left-1/2 top-1/2 m-auto w-96 h-48 -ml-48 -mt-24 flex flex-col items-center justify-center"
  >
    <img
      alt="Neutral illustration"
      class="opacity-80 mb-2.5"
      src="/neutral.png"
      width="80"
    />
    <p>
      No results to display.
    </p>
  </div>
</DocumentFragment>
`;

exports[`Search View matches the snapshot when error 1`] = `
<DocumentFragment>
  <header
    class="w-full h-12 flex items-center p-2 md:hidden justify-between"
  >
    <button
      class="w-11 float-left"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-arrow-left fa-w-14 m-1 text-lg"
        data-icon="arrow-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
          fill="currentColor"
        />
      </svg>
    </button>
    <h1
      class="mx-2 font-medium text-lg overflow-x-auto whitespace-nowrap"
    >
      Error
    </h1>
    <div
      class="w-11"
    />
  </header>
  <div
    class="px-2 py-2.5 bg-red-400 bg-opacity-50 border border-solid border-red-400 p-2.5"
  >
    <div
      class="flex flex-col items-center justify-center"
    >
      <img
        alt="Frustrated Illustration"
        class="opacity-80 mb-2.5"
        src="/frustrated.png"
        width="80"
      />
    </div>
    <h2
      class="font-bold"
    >
      Error
    </h2>
    <p>
      Generic error
    </p>
  </div>
</DocumentFragment>
`;

exports[`Search View matches the snapshot when loading 1`] = `
<DocumentFragment>
  <header
    class="w-full h-12 flex items-center p-2 md:hidden justify-between"
  >
    <button
      class="w-11 float-left"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-arrow-left fa-w-14 m-1 text-lg"
        data-icon="arrow-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
          fill="currentColor"
        />
      </svg>
    </button>
    <h1
      class="mx-2 font-medium text-lg overflow-x-auto whitespace-nowrap"
    >
      Search
    </h1>
    <div
      class="w-11"
    />
  </header>
  <div
    class="px-2"
  >
    <input
      aria-label="search-input"
      class="w-full mb-4 p-1 text-xl text-center bg-gray-200 dark:bg-gray-600 border border-solid border-gray-800 focus:border-gray-400"
      placeholder="Search..."
      type="search"
      value=""
    />
  </div>
  <div
    class="flex flex-row flex-wrap px-2"
  >
    <div
      class="w-1/3 sm:w-1/4 md:w-1/5 p-1"
    >
      <div
        class="animate-pulse bg-gray-700 w-full h-0 relative bg-gray-800"
        style="padding-top: 140.47619047619045%;"
      />
      <div
        class="animate-pulse bg-gray-700 w-full h-16 bg-gray-800 p-2"
      />
    </div>
    <div
      class="w-1/3 sm:w-1/4 md:w-1/5 p-1"
    >
      <div
        class="animate-pulse bg-gray-700 w-full h-0 relative bg-gray-800"
        style="padding-top: 140.47619047619045%;"
      />
      <div
        class="animate-pulse bg-gray-700 w-full h-16 bg-gray-800 p-2"
      />
    </div>
    <div
      class="w-1/3 sm:w-1/4 md:w-1/5 p-1"
    >
      <div
        class="animate-pulse bg-gray-700 w-full h-0 relative bg-gray-800"
        style="padding-top: 140.47619047619045%;"
      />
      <div
        class="animate-pulse bg-gray-700 w-full h-16 bg-gray-800 p-2"
      />
    </div>
    <div
      class="w-1/3 sm:w-1/4 md:w-1/5 p-1"
    >
      <div
        class="animate-pulse bg-gray-700 w-full h-0 relative bg-gray-800"
        style="padding-top: 140.47619047619045%;"
      />
      <div
        class="animate-pulse bg-gray-700 w-full h-16 bg-gray-800 p-2"
      />
    </div>
    <div
      class="w-1/3 sm:w-1/4 md:w-1/5 p-1"
    >
      <div
        class="animate-pulse bg-gray-700 w-full h-0 relative bg-gray-800"
        style="padding-top: 140.47619047619045%;"
      />
      <div
        class="animate-pulse bg-gray-700 w-full h-16 bg-gray-800 p-2"
      />
    </div>
    <div
      class="w-1/3 sm:w-1/4 md:w-1/5 p-1"
    >
      <div
        class="animate-pulse bg-gray-700 w-full h-0 relative bg-gray-800"
        style="padding-top: 140.47619047619045%;"
      />
      <div
        class="animate-pulse bg-gray-700 w-full h-16 bg-gray-800 p-2"
      />
    </div>
    <div
      class="w-1/3 sm:w-1/4 md:w-1/5 p-1"
    >
      <div
        class="animate-pulse bg-gray-700 w-full h-0 relative bg-gray-800"
        style="padding-top: 140.47619047619045%;"
      />
      <div
        class="animate-pulse bg-gray-700 w-full h-16 bg-gray-800 p-2"
      />
    </div>
    <div
      class="w-1/3 sm:w-1/4 md:w-1/5 p-1"
    >
      <div
        class="animate-pulse bg-gray-700 w-full h-0 relative bg-gray-800"
        style="padding-top: 140.47619047619045%;"
      />
      <div
        class="animate-pulse bg-gray-700 w-full h-16 bg-gray-800 p-2"
      />
    </div>
    <div
      class="w-1/3 sm:w-1/4 md:w-1/5 p-1"
    >
      <div
        class="animate-pulse bg-gray-700 w-full h-0 relative bg-gray-800"
        style="padding-top: 140.47619047619045%;"
      />
      <div
        class="animate-pulse bg-gray-700 w-full h-16 bg-gray-800 p-2"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Search View matches the snapshot with data 1`] = `
<DocumentFragment>
  <header
    class="w-full h-12 flex items-center p-2 md:hidden justify-between"
  >
    <button
      class="w-11 float-left"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-arrow-left fa-w-14 m-1 text-lg"
        data-icon="arrow-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
          fill="currentColor"
        />
      </svg>
    </button>
    <h1
      class="mx-2 font-medium text-lg overflow-x-auto whitespace-nowrap"
    >
      Search
    </h1>
    <div
      class="w-11"
    />
  </header>
  <div
    class="px-2"
  >
    <input
      aria-label="search-input"
      class="w-full mb-4 p-1 text-xl text-center bg-gray-200 dark:bg-gray-600 border border-solid border-gray-800 focus:border-gray-400"
      placeholder="Search..."
      type="search"
      value=""
    />
  </div>
  <div
    class="flex flex-row flex-wrap px-2"
  >
    <div
      class="w-1/3 sm:w-1/4 md:w-1/5 p-1"
    >
      <a
        class="flex flex-col bg-gray-200 dark:bg-gray-800 h-full border-b-2 border-solid border-green-500 shadow"
        href="/show/216"
      >
        <div
          class="h-0 relative box-border p-1 border-2 border-gray-200 dark:border-gray-800"
          style="padding-top: 140.47619047619045%;"
        >
          <div
            class="animate-pulse bg-gray-700 absolute top-0 left-0 w-full h-full bg-gray-800"
          />
          <img
            alt="Rick and Morty"
            class="absolute top-0 left-0 w-full h-full hidden"
            src="undefined/uploads/images/medium_portrait/1/3603.jpg"
          />
        </div>
        <div
          class="min-h-16 flex-grow flex flex-col justify-around items-center p-2"
        >
          <p
            class="text-center"
          >
            Rick and Morty
          </p>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Search View matches the snapshot with no data 1`] = `
<DocumentFragment>
  <header
    class="w-full h-12 flex items-center p-2 md:hidden justify-between"
  >
    <button
      class="w-11 float-left"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-arrow-left fa-w-14 m-1 text-lg"
        data-icon="arrow-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
          fill="currentColor"
        />
      </svg>
    </button>
    <h1
      class="mx-2 font-medium text-lg overflow-x-auto whitespace-nowrap"
    >
      Search
    </h1>
    <div
      class="w-11"
    />
  </header>
  <div
    class="px-2"
  >
    <input
      aria-label="search-input"
      class="w-full mb-4 p-1 text-xl text-center bg-gray-200 dark:bg-gray-600 border border-solid border-gray-800 focus:border-gray-400"
      placeholder="Search..."
      type="search"
      value=""
    />
  </div>
  <div
    class="absolute left-1/2 top-1/2 m-auto w-80 h-96 -ml-40 -mt-48 flex flex-col items-center justify-center"
  >
    <img
      alt="Frustrated illustration"
      class="opacity-80 mb-2.5"
      src="/frustrated.png"
      width="80"
    />
    <p>
      No shows match your search query.
    </p>
  </div>
</DocumentFragment>
`;
